template/server.js

```diff
@@ -1,106 +1,110 @@
 
-  return {
-    head: template.slice(0, index),
-    tail: template.slice(index + applicationHTML.length),
-  }
-})();
+const app = express();
 
 // create renderer with createBundleRenderer
 // and a LRU cache of 15 mins with max capacity of 1000
-function createRenderer (bundle) {
-  return createBundleRenderer(bundle, {
-    cache: require('lru-cache')({
+function createRenderer (bundle, options) {
+  return createBundleRenderer(bundle, Object.assign(options, {
+    cache: LRU({
       max: 1000,
       maxAge: 1000 * 60 * 15,
     }),
-  });
+    basedir: resolve('./dist'),
+    runInNewContext: false,
+  }))
 }
 
-// production renderer works off of a pre-built bundle
-// development renderer works off of a bundle that's being re-built on every change
-// both bundles are built with webpack
 let renderer;
-if (isProduction) {
-  renderer = createRenderer(fs.readFileSync(
-    resolve('./dist/server-bundle.js'), 
-    'utf-8'
-  ));
-} else {
-  require('./build/dev-server')(application, bundle => {
-    renderer = createRenderer(bundle);
+let readyPromise;
+const templatePath = resolve('./src/index.template.html');
+if (isProd) {
+  const template = fs.readFileSync(templatePath, 'utf-8');
+  const bundle = require('./dist/vue-ssr-server-bundle.json');
+  const clientManifest = require('./dist/vue-ssr-client-manifest.json');
+
+  renderer = createRenderer(bundle, {
+    template,
+    clientManifest,
   });
+} else {
+  readyPromise =  require('./build/setup-dev-server')(
+    app,
+    templatePath,
+    (bundle, options) => {
+      renderer = createRenderer(bundle, options);
+    }
+  )
 }
 
-application.use(favicon(resolve('./static/favicon.ico')));
-if(isProduction) {
-  application.use('/static', express.static(resolve('./dist/static')));
-}
+const serve = (path, cache) => express.static(resolve(path), {
+  maxAge: cache && isProd ? 1000 * 60 * 60 * 24 * 30 : 0,
+});
 
-// writes the head of index.html
-// then injects the initial state of the application
-// then writes the application as it's being rendered
-// finally writes the tail of index.html
-application.get('*', (request, response) => {
-  if (!renderer) {
-    return response.end('Compiling... Refresh in a moment!');
+app.use(compression({ threshold: 0  }));
+app.use(favicon(resolve('./public/favicon.ico')));
+app.use('/dist', serve('./dist', true));
+app.use('/public', serve('./public', true));
+app.use('/manifest.json', serve('./manifest.json', true));
+app.use('/service-worker.js', serve('./dist/service-worker.js'));
+
+app.use(microcache.cacheSeconds(1, req => useMicroCache && req.originalUrl));
+
+function render (req, res) {
+  const s = Date.now();
+
+  res.setHeader("Content-Type", "text/html");
+  res.setHeader("Server", serverInfo);
+
+  const handleError = err => {
+    if (err.url) {
+      res.redirect(err.url)
+    } else if (err.code === 404) {
+      res.status(404).send('404 | Page Not Found');
+    } else {
```

Comment:
I suggest setting up templates for custom non verbose 404/500 errors - clients often expect them designed out.